apps/scan/src/main/scala/org/lfdecentralizedtrust/splice/scan/store/ScanEventStore.scala

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package org.lfdecentralizedtrust.splice.scan.store
+
+import com.digitalasset.canton.logging.NamedLoggerFactory
+import com.digitalasset.canton.logging.NamedLogging
+import com.digitalasset.canton.tracing.TraceContext
+import org.lfdecentralizedtrust.splice.scan.store.db.DbScanVerdictStore
+import org.lfdecentralizedtrust.splice.store.TreeUpdateWithMigrationId
+import org.lfdecentralizedtrust.splice.store.UpdateHistory
+import com.digitalasset.canton.data.CantonTimestamp
+import org.lfdecentralizedtrust.splice.store.PageLimit
+
+import scala.concurrent.{ExecutionContext, Future}
+
+/** Combines data from UpdateHistory store and DbScanVerdictStore, for events endpoints */
+class ScanEventStore(
+    val verdictStore: DbScanVerdictStore,
+    val updateHistory: UpdateHistory,
+    protected val loggerFactory: NamedLoggerFactory,
+)(implicit ec: ExecutionContext)
+    extends NamedLogging {
+
+  type VerdictT = DbScanVerdictStore.VerdictT
+  type TransactionViewT = DbScanVerdictStore.TransactionViewT
+  type Verdict = (VerdictT, Seq[TransactionViewT])
+  type Event = (Option[Verdict], Option[TreeUpdateWithMigrationId])
+
+  def getEventByUpdateId(updateId: String)(implicit tc: TraceContext): Future[Option[Event]] = {
+    val fUpdate = updateHistory.getUpdate(updateId)
+    val fVerdict = verdictStore.getVerdictByUpdateId(updateId)
+    for {
+      updateO <- fUpdate
+      verdictO <- fVerdict
+      verdictWithViewsO <- verdictO match {
+        case Some(v) => verdictStore.listTransactionViews(v.rowId).map(views => Some((v, views)))
+        case None => Future.successful(None)
+      }
+    } yield {
+      if (updateO.isEmpty && verdictWithViewsO.isEmpty) None else Some((verdictWithViewsO, updateO))
+    }
+  }
+
+  /** Simple Algo, useful for testing
+    */
+  def getEventsReference(
```

Comment:
This has been used as a reference to verify the logic for the unit tests. I will add the proper SQL based API for doing the actual query and add tests to verify it matches the output from this reference.